doc/LIMITS.md

```diff
@@ -0,0 +1,251 @@
+
+      /**
+       * The total number of outbound protocol streams that can be opened on a given connection
+       */
+      maxOutboundStreams: 1024
+    })
+  ]
+})
+```
+
+### Protocol limits
+
+When registering listeners for custom protocols, the maximum number of simultaneously open inbound and outbound streams per-connection can be specified. If not specified these will default to 32 inbound streams and 64 outbound streams.
+
+If more than this number of streams for the given protocol are opened on a single connection, subsequent new streams for that protocol will be immediately reset.
+
+Since incoming stream data is buffered until it is comsumed, you should attempt to specify the minimum amount of streams required to keep memory usage to a minimum.
+
+```js
+libp2p.handle('/my-protocol/1.0.0', (streamData) => {
+  // ..handle stream
+}, {
+  maxInboundStreams: 10, // defaults to 32
+  maxOutboundStreams: 10, // defaults to 64
+})
+```
+
+## Closing connections
+
+When choosing connections to close the connection manager sorts the list of connections by the value derived from the tags given to each peer. The values of all tags are summed and connections with lower valued peers are elibible for closing first.
+
+```js
+// tag a peer
+await libp2p.peerStore.tagPeer(peerId, 'my-tag', {
+  value: 50, // 0-100 is the typical value range
+  ttl: 1000 // optional field, this tag will be deleted after this many ms
+})
+```
+
+## Transport specific limits
+
+Some transports allow configuring additional limits, please see their READMEs for full config options.
+
+A non-exhaustive list follows:
+
+### TCP
+
+The [@libp2p/tcp](https://github.com/libp2p/js-libp2p-tcp) transport allows additional limits to be configured
+
+```js
+const node = await createLibp2pNode({
+  transports: [
+    new TCP({
+      /**
+       * Inbound connections with no activity in this timeframe (ms) will be closed
+       */
+      inboundSocketInactivityTimeout: 30000,
+
+      /**
+       * Outbound connections with no activity in this timeframe (ms) will be closed
+       */
+      outboundSocketInactivityTimeout: 60000,
+
+      /**
+       * Once this many connections are open on this listener any further connections
+       * will be rejected - this will have no effect if it is larger than the value
+       * configured for the ConnectionManager maxConnections parameter
+       */
+      maxConnections: 200
+    })
+  ]
+})
+```
+
+## Allow/deny lists
+
+It is possible to configure some hosts to always accept connections from and some to always reject connections from.
+
+```js
+const node = await createLibp2pNode({
+  connectionManager: {
+    /**
+     * A list of multiaddrs, any connection with a `remoteAddress` property
+     * that has any of these addresses as a prefix will be accepted ignoring
+     * all connection limits
+     */
+    allow: [
+      '/ip4/43.123.5.23/tcp/3984',
+      '/ip4/234.243.64.2',
+      '/ip4/52.55',
```

Comment:
This isn't a valid multiaddr, isn't that confusing? I think we should prefer to use the standard IP cidr format (I know this is just the docs, so this will involve impl changes) e.g. "/ip4/52.55" should be "/ip4/52.55.0.0/ipcidr/16".

Reply:
They are treated as strings prefixes not full multiaddrs.  It'd probably be more correct to make them multiaddrs and less error prone but I don't think the `@multiformats/multiaddr` module will let us match address ranges.

A good improvement for the future.